# <table border=0 width="100%"><tr><td><p align="left"><img src="..\img\logo-1.jpg" align="left" width=300></p></td><td><font size=3><B>Data analysis (Python) - homework for week 9</B></font></td></tr></table>

tips 
- rename the name jupyter notebook file following the rules: **replace ID and name with your student ID and your own name**

In [ ]:
name = input("what is your name: ")

In [ ]:
student_ID = input("what is your student ID: ")

I loaded the data of The Shanghai composite index saved in file "000001_trade.csv". The columns of the table are: High,Low,Open,Close,Volume,Adj Close

your task - please get the count of the days on which return is positive (i.e., the colse price increase compared to yesterday)

tips - 
* your can try the function of np [numpy.diff](https://docs.scipy.org/doc/numpy/reference/generated/numpy.diff.html)

In [29]:
import numpy as np

In [30]:
# read data from csv file
sh_path = "000001_trade.csv"
data = np.loadtxt(sh_path,delimiter=",",skiprows= 1,usecols=(1,2,3,4,5,6),dtype =float) #没有读取日期

In [31]:
data

array([[  3295.27905273,   3243.3190918 ,   3289.75      ,
          3243.76000977, 109400.        ,   3243.76000977],
       [  3290.51196289,   3221.46191406,   3254.46801758,
          3282.17895508, 126200.        ,   3282.17895508],
       [  3295.86791992,   3253.04394531,   3277.51708984,
          3254.21508789, 123600.        ,   3254.21508789],
       ...,
       [  2861.13110352,   2824.35498047,   2833.01489258,
          2852.12988281, 166600.        ,   2852.12988281],
       [  2927.42993164,   2854.07006836,   2854.07006836,
          2925.71606445, 260500.        ,   2925.71606445],
       [  2924.7019043 ,   2903.88305664,   2917.22412109,
          2909.37988281, 240300.        ,   2909.37988281]])

In [32]:
len(data)

2292

# np.diff函数介绍实例

In [33]:
a = np.array([1,2,4,2]) 

# print(len(a))

# np.diff(a)

# np.diff(a)>0

sum(np.diff(a)>0)


2

# 开始解题

In [34]:
# please stard code here 

# 使用一行代码就能解决

#np.diff(data[:,3])

#np.diff(data[:,3])>0

sum(np.diff(data[:,3])>0)

1200

In [35]:
# please stard code here 

# 另一种思路，也可以使用多行代码（逻辑：先计算布尔值，再进行匹配）

data_pool = np.diff(data[:,3])>0

data_pool



array([ True, False, False, ...,  True,  True, False])

In [36]:
len(data_pool)

2291

In [37]:
data = np.delete(data,0,axis = 0) # 去除原始数据的第一行，即第一天

positive_day = len(data[data_pool]) # 返回布尔值为ture的数组长度

positive_day

1200

IndexError: boolean index did not match indexed array along dimension 0; dimension is 2292 but corresponding boolean dimension is 2291


**self-leaning materials after class**

* read more about numpy in chapter 2 Introduction to NumPy (T1 - Python_Data_Science_Handbook)

* Preview pandas which is a very import tool for Data Manipulation - 
**T1 – chp 3 Data Manipulation with Pandas**